src/ngAnimate/animateCss.js

```diff
@@ -665,7 +665,7 @@ var $AnimateCssProvider = ['$animateProvider', function($animateProvider) {
       // we need to recalculate the delay value since we used a pre-emptive negative
       // delay value and the delay value is required for the final event checking. This
       // property will ensure that this will happen after the RAF phase has passed.
-      if (timings.transitionDuration > 0) {
+      if (options.duration == null && timings.transitionDuration > 0) {
```

Comment:
This line seems to be sufficient to cause the test to pass. Are the other lines (i.e. `maxDelay = Math.max(relativeDelay, 0) || 0;`) unnecessary?